File: frontend/src/Books/Book.ts
import {Author} from "./Author";

export type Book = {
  readonly id: number;
  readonly name: string;
  readonly published: string;
  readonly language: string;
  readonly subject: string;
  readonly publisher: string;
  readonly cover: string;
  readonly book: string;
  readonly author: Author[];
  readonly collectionId: number;
}

File: backend/dto/book.go
package dto

import (
	"time"
)

type Book struct {
	ID              uint       `json:"id"`
	Title           string     `json:"title"`
	Published       *time.Time `json:"published"`
	Language        string     `json:"language"`
	Subjects        []Subject  `json:"subjects"`
	Publisher       *string    `json:"publisher"`
	Cover           *[]byte    `json:"cover"`
	Book            string     `json:"book"`
	Authors         []Author   `json:"authors"`
	CollectionId    *uint      `json:"collectionId"`
	CollectionIndex *uint      `json:"collectionIndex"`
}
